foreman_stub.py

```diff
@@ -9,61 +9,146 @@
 from collections import defaultdict
 from urllib.parse import urlparse
 
-from flask import Flask, jsonify, request
+from flask import Flask, abort, jsonify, request
 
 
 DEBUG = os.getenv('DEBUG', 'false').lower() not in ('false', '0', '', 'off')
 
 if DEBUG:
-    import pip
-    pip.main(['install', 'q', 'epdb'])
+    try:
+        import epdb
+    except ImportError:
+        pass
+    else:
+        epdb.serve()
 
-app = Flask(__name__)
+app = Flask(__name__)  # pylint: disable=invalid-name
 
 PAGECACHE = defaultdict(dict)
+REPORTS = []
 
 
 @app.before_first_request
 def build_pagecache():
     """Read JSON fixtures into in-memory cache."""
     jfiles = glob.glob('fixtures/*.json')
     for jfile in jfiles:
-        with open(jfile, 'r') as f:
-            jdata = json.load(f)
+        with open(jfile, 'r') as fixture:
+            jdata = json.load(fixture)
         key = urlparse(jdata['url']).path[7:]
         subkey = jdata['page']
         PAGECACHE[key][subkey] = jdata['data']
 
+        if 'id' in jdata['data'] and key == f"/hosts/{jdata['data']['id']}":
```

Comment:
Pythonic way would be to follow EAFP with try/except KeyError

Reply:
Right now you can got to `/hosts`, find a host and `/hosts/ID` will return a 404 which meant it took me some time to find a valid host when testing in the callback plugin.

This is an ugly workaround to have both `/hosts/1` and `/hosts/host.example.com` work but it's really a partial solution. Looking at the data set it might be easier to have a plain list of hosts and pretend it's a database.

What do you think about changing the `PAGECACHE` to `HOSTS = []` and appending all results? I'm already imitating the pagination elsewhere anyway.

Reply:
I think that refactoring and actual changes should be separated. I cannot review this when there's more unrelated changes than justified ones.

Reply:
As for the check, I assume you meant something like
```suggestion
        if key == f"/hosts/{jdata['data'].get('id') or ''}":
```

Reply:
I think that's incorrect because if there's no ID, there is probably also no hostname. Otherwise I think you'll be mapping lists to individual data pages.